#!/usr/bin/env python
PACKAGE = "ur5_control"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("Virt_Comp", bool_t, 0, "Activate virtual compliance to external forces", False)
gen.add("Kf_ext_xy", double_t, 0, "Scaling of external force effects on virtual compliance", 0.0, 0.0, 1.0)
gen.add("Kf_ext_z", double_t, 0, "Scaling of external force effects on virtual compliance", 0.0, 0.0, 1.0)
#
gen.add("d_thresh", double_t, 0, "Distance threshold between the EE and the DMP's current goal [m]", 0.15, 0.0, 1.0)
#
gen.add("Kcd", double_t, 0, "Gain of hand distance damping effect - Time constant coupling", 0.0, 0.0, 1000.0)
gen.add("Ktd", double_t, 0, "Gain of hand distance damping effect - Spatial coupling", 0.0, 0.0, 1000.0)
gen.add("Kce", double_t, 0, "Gain of tracking error spring effect - Time constant coupling", 0.0, 0.0, 100.0)
gen.add("Kte", double_t, 0, "Gain of tracking error spring effect - Spatial coupling", 0.0, 0.0, 1000.0)
#
gen.add("sigm_steep", double_t, 0, "Hand distance damping effect - Sigmoid steepness", 1.0, -100.0, 100.0)
gen.add("sigm_off", double_t, 0, "Hand distance damping effect - Sigmoid offset", 0.0, -50.0, 50.0)

exit(gen.generate(PACKAGE, "ur5_control", "DMPExternalEffects"))
